python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -0,0 +1,620 @@
+cimport cuml.common.cuda
+
+cdef extern from "randomforest/randomforest.hpp" namespace "ML":
+    cdef enum CRITERION:
+        GINI,
+        ENTROPY,
+        MSE,
+        MAE,
+        CRITERION_END
+
+cdef extern from "decisiontree/decisiontree.hpp" namespace "ML::DecisionTree":
+    cdef struct DecisionTreeParams:
+        int max_depth
+        int max_leaves
+        float max_features
+        int n_bins
+        int split_algo
+        int min_rows_per_node
+        bool bootstrap_features
+        bool quantile_per_tree
+        CRITERION split_criterion
+
+cdef extern from "randomforest/randomforest.hpp" namespace "ML":
+
+    cdef enum RF_type:
+        CLASSIFICATION,
+        REGRESSION
+
+    cdef struct RF_metrics:
+        RF_type rf_type
+        float accuracy
+        double mean_abs_error
+        double mean_squared_error
+        double median_abs_error
+
+    cdef struct RF_params:
+        int n_trees
+        bool bootstrap
+        float rows_sample
+        pass
+
+    cdef cppclass RandomForestMetaData[T, L]:
+        void* trees
+        RF_params rf_params
+
+    # Random Forest Classifier
+    cdef void fit(cumlHandle & handle,
+                  RandomForestMetaData[float, float]*,
+                  float *,
+                  int,
+                  int,
+                  float *,
+                  RF_params) except +
+
+    cdef void fit(cumlHandle & handle,
+                  RandomForestMetaData[double, double]*,
+                  double *,
+                  int,
+                  int,
+                  double *,
+                  RF_params) except +
+
+    cdef void predict(cumlHandle& handle,
+                      RandomForestMetaData[float, float] *,
+                      float *,
+                      int,
+                      int,
+                      float *,
+                      bool) except +
+
+    cdef void predict(cumlHandle& handle,
+                      RandomForestMetaData[double, double]*,
+                      double *,
+                      int,
+                      int,
+                      double *,
+                      bool) except +
+
+    cdef RF_metrics score(cumlHandle& handle,
+                          RandomForestMetaData[float, float]*,
+                          float *, float *,
+                          int, int, float *, bool) except +
+
+    cdef RF_metrics score(cumlHandle& handle,
+                          RandomForestMetaData[double, double]*,
+                          double *, double *,
+                          int, int, double *, bool) except +
+
+    cdef void print_rf_summary(RandomForestMetaData[float, float]*) except +
+    cdef void print_rf_summary(RandomForestMetaData[double, double]*) except +
+
+    cdef void print_rf_detailed(RandomForestMetaData[float, float]*) except +
+    cdef void print_rf_detailed(RandomForestMetaData[double, double]*) except +
+
+    cdef RF_params set_rf_class_obj(int, int, float,
```

Comment:
Same as above, one line per parameter makes it easier to check and spot for errors